go/tools/builders/compile.go

```diff
@@ -19,30 +19,45 @@ package main
 import (
 	"fmt"
 	"go/build"
+	"go/parser"
+	"go/token"
 	"log"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
+	"strings"
 )
 
 func run(args []string) error {
 	// process the args
 	if len(args) < 2 {
-		return fmt.Errorf("Usage: compile gotool [sources] -- <extra options>")
+		return fmt.Errorf("Usage: compile gotool [-dep importpath ...] [sources] -- <extra options>")
 	}
 	gotool := args[0]
 	args = args[1:]
 	sources := []string{}
 	goopts := []string{}
+	deps := []string{}
 	bctx := build.Default
 	bctx.CgoEnabled = true
+	depNext := false
+argsLoop:
 	for i, s := range args {
```

Comment:
Good idea. I reused multiFlag in flags.go, which is already used for the linker.